x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/extractor/ProcessedField.java

```diff
@@ -24,20 +27,21 @@ public ProcessedField(PreProcessor processor) {
     }
 
     public List<String> getInputFieldNames() {
-        return preProcessor.inputFields();
+        return preProcessor.inputFields().stream().sorted().collect(Collectors.toList());
```

Comment:
This is called for every document we load into the process. I suggest we sort once when we store them in the `PreProcessor`.